### ISSUE TITLE
standalone node js to start stream on motion or ding

### ISSUE THREAD

**Issue by privatesam** [DustinBryant/MMM-Ring#3]
Hi thanks for this work its really useful for something I'm trying to achieve.  I saw you conversation over [here](https://github.com/dgreif/ring/issues/117) at dgreif's repo and think you might be able to help me to achieve what I'm trying to do with dgreif's ring-api.

Basically I have successfully ran dgreif's [example.ts](https://github.com/dgreif/ring/blob/master/examples/example.ts) to actively listen for motion and dings.  I can also successfully run the [browser-example.ts ](https://github.com/dgreif/ring/blob/master/examples/browser-example.ts)example to get an active m3u8 stream.  However I'm not well versed in node and tried mash these two examples together so in a simple standalone node script which listens for motion and dings and when one is received start up the sipSession m3u8 stream.  However I've run into a few issues which I think are similar to what you raised the issue about over there.

This is my clunky mash-up [script](https://gist.github.com/privatesam/721f117bb6f2a018cc8496cd61246e27).  It sort of works but has issues with paused video on the stream and I think it calls a new stream if multiple motion alerts are received.  Basically its badly written.

I've tried looking through your [node_helper.js](https://github.com/DustinBryant/MMM-Ring/blob/master/node_helper.js) which I think has pretty much all the code I need to achieve my aim but its obviously designed to run with the MagicMirror framework as an add-on.  Is there an easy enough tweak to run this as a standalone node app to simply fire up an m3u8 stream upon motion or ding?  

The reason this would be useful to me and indeed many others is we can then point our NAS/security camera software at the m3u8 stream and get it to record into our existing security set ups.  

Thanks for any help you can provide.

**Comment by DustinBryant** [comment]
Sounds like a cool usage case.

Here are a few things I noticed with ring-api by dgreif. Whenever there is an active motion/ring event and you use the ring-api to start your own it instead piggy backs onto the existing event. That's to say, it won't create its own event/stream for your call to start one. This also means, that event will show up as answered in the Ring App/Website.

Are you just trying to store the stream for later? Or, are you trying to view the stream through some type of browser?

Your "pause" issue is actually related to a browser setting. A while back it was decided that videos with sound are not allowed to auto-play or be told to play via js DOM. You can set the Video html element attribute muted which will play the video without sound.

Or, you can find the flag in whatever browser you use (chrome example) **autoplay-policy** and set it to **no-user-gesture-required**. If you're using something like Electron then you'll need to apply that same flag/rule within its configuration on the ready event and using commandLine.appendSwitch("autoplay-policy", "no-user-gesture-required");.

Let me know if I missed anything. Not entirely sure what issue you're experiencing other than the pausing.

**Comment by privatesam** [comment]
Thanks for the reply on this and for clarifying the way the stream works.  I'm trying to iron various bugs in my node script at the moment.  Looking through your code is there a reason you clear out all the video files in the video folder after a stream finishes?  Also did you select not to start a stream up on motion just to keep down the amount of times the stream is grabbed?

**Comment by DustinBryant** [comment]
Technically, the clearing out of videos is unnecessary. The options for ffmpeg -hls_flags delete_segments basically automatically does that work for you. I put it in place for precautionary measures just in case something happens and potentially multiple video segment gets orphaned and not picked up by that process. I specifically had to keep storage in mind because this is ran on Raspberry Pi's where storage is typically on an SD Card and can range in size significantly. I'll admit it is a bit overkill.

Though I do have a Ring stick-up cam, I only cared about the doorbell video. The MagicMirror I have is nearest to the entry of my home, and it was set up so that when I start to head to the door I can instantly see who is ringing the doorbell before answering. Also, I've found there to be a lot of false-positives with motion and I didn't want the mirror to basically always be showing video due to that. I felt it necessary only for the doorbell events.
